voting/src/contract.rs

```diff
@@ -354,12 +359,9 @@ fn locked_amount<S: Storage, A: Api, Q: Querier>(voter: &CanonicalAddr, deps: &m
     let voter_key = &voter.as_slice();
     let token_manager = bank_read(&deps.storage).load(voter_key).unwrap();
 
-    token_manager.participated_polls.iter().for_each(| poll_id | {
-        let poll_key = get_poll_voter_key(*poll_id, &voter);
-        let voter_info = poll_voter_info_read(&deps.storage).load(poll_key.as_bytes()).unwrap();
-
-        if voter_info.weight.u128() > largest {
-            largest = voter_info.weight.u128();
+    token_manager.locked_tokens.values().for_each(| weight | {
```

Comment:
Okay, I see why you do that here, you want to make this simple to iterate over without needing to read 10 keys when there are 10 polls.

Two points: you can use [max](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max) to simplify this code:

`let largest = token_manager.locked_tokens.values().max().unwrap_or_default();`

Second, you could store `Vec<(u64, Uint128)>` as the type if you are going to iterate over all of them. That is serializable for sure and gives you the same power (except direct lookup is a bit more expensive with `find`, but if the list if large enough for that to be a problem, you should use unique keys in the db.

With this, you could do

`let largest = token_manager.locked_tokens.iter().map(|(_, v)| v).max().unwrap_or_default();`

Sure, `iter().map(|(_, v)| v)` is a bit longer than `values()` but close